### DISCUSSION TITLE
Shorthand syntax for a slice, map, string literals allocated with the temp allocator (or main allocator)

### DISCUSSION THREAD

**Discussion by Beefster09** [odin-lang/Odin#3761] [Ideas/Requests]
Currently you can sidestep this with `slice.clone`, `strings.clone`, and `???.clone` around a stack-allocated literal (which allows the value to be returned), but this is long, awkward, and verbose. Would be nice to have something like this:

```
foo :: proc(x: int) -> map[string][]int {
    return &[context.temp_allocator] {  // any expression evaluating to an allocator can go here
        "bar" = {x + 1, 2, 3},  // the allocator is applied to the string and slice literals as well
        "baz" = {4, 5},
    }
}
```

and perhaps there could also be a shorthand for `context.temp_allocator` and `context.allocator`, perhaps `#t` and `#a`? Maybe only for this context?

**Comment by Feoramund** [comment]
```odin
my_map: map[string]i8
{
	context.allocator = context.temp_allocator
	my_map = {"hellope" = 3}
}
```

This is fine if you need to do it once. If you don't need the primary allocator past a certain point, you can assign it to the temporary one from then on.

```odin
context.allocator = context.temp_allocator
my_map := map[string]i8 {"hellope" = 3}
my_other_map := map[string]i8 {"hellope" = 3}
```
